File: javapatterns/src/main/java/org/iut/nantes/adapter/chimie/NewDB.java
package org.iut.nantes.adapter.chimie;

import java.util.HashMap;

class NewDB
{
  private HashMap<String, Molecule> database_;

  public NewDB()
  {
    database_ = new HashMap<String, Molecule>();
  }

  public void newAdd( Molecule m ) { database_.put( m.getName(), m ); }
  public boolean exists( String name ) { return database_.containsKey( name ); }
  public Molecule newMolecule( String name ) { return database_.get( name ); }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/chimie/NewDB.kt
package org.iut.nantes.adapter.chimie

import java.util.*

internal class NewDB {
    private val database_: HashMap<String?, Molecule>
    fun newAdd(m: Molecule) {
        database_[m.name] = m
    }

    fun exists(name: String?): Boolean {
        return database_.containsKey(name)
    }

    fun newMolecule(name: String?): Molecule? {
        return database_[name]
    }

    init {
        database_ = HashMap()
    }
}